pkg/kubehound/storage/graphdb/janusgraph_provider.go

```diff
@@ -70,19 +70,47 @@ func (jgp *JanusGraphProvider) Prepare(ctx context.Context) error {
 	tx := g.Tx()
 	defer tx.Close()
 
-	gtx, err := tx.Begin()
-	if err != nil {
-		return err
-	}
-
-	err = <-gtx.V().Drop().Iterate()
-	if err != nil {
-		return err
-	}
-
-	err = tx.Commit()
-	if err != nil {
-		return err
+	for {
+		// Begin a new transaction.
+		gtx, err := tx.Begin()
+		if err != nil {
+			return err
+		}
+
+		// Retrieve the number of vertices in the graph.
+		page, err := gtx.V().Count().Next()
+		if err != nil {
+			return err
+		}
+
+		// Decode the number of vertices from the page.
+		count, err := page.GetInt()
+		if err != nil {
+			return err
+		}
+
+		// If there are no more vertices to delete, break the loop.
+		if count == 0 {
+			break
+		}
+
+		// Delete the vertices in the graph.
+		err = <-gtx.V().Limit(10000).Drop().Iterate()
```

Comment:
Nit: worth extracting that const into a global ? (and, later on maybe provide that as a config)